### PR TITLE
Homepage

### PR DESCRIPTION
**Pull request by JayceEse**

### CODE REVIEW

File: src/Pages/kitchen_sink/KitchenSink.tsx [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+    { img: CrystalCaves, label: "The Crystal Caves" },
+  ];
+
+  interface ImgSlideStyle {
+    backgroundImage: string;
+    backgroundSize: string;
+    backgroundRepeat: string;
+    backgroundPosition: string;
+    width: string;
+    height: string;
```
Comment (JayceDugan) [inline src/Pages/kitchen_sink/KitchenSink.tsx:53]: Optional / curiosity: Are you able to type these to the actual CSS property types?

Comment (JayceDugan) [reply src/Pages/kitchen_sink/KitchenSink.tsx:53]:
```
interface ImgSlideStyle {
  backgroundSize: BackgroundSize // "cover" "repeat" "contain"
}
```

File: src/Pages/kitchen_sink/KitchenSink.tsx [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+  return (
+    <>
+      <div
+        style={imgSlide}
+        className="flex flex-col justify-between duration-500"
+      >
+        {/* Destination Info */}
+        <div className="ml-24 mt-20">
+          <div className="text-5xl font-noto text-white font-bold w-72 h-28 mb-8">
+            <h1>{data[currentIndex].label}</h1>
```
Comment (JayceDugan) [inline src/Pages/kitchen_sink/KitchenSink.tsx:130]: Nit / optional: Would maybe move this to a variable.

File: src/Pages/kitchen_sink/KitchenSink.tsx [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+          </div>
+          <div className="flex flex-row justify-between w-80">
+            <p className="font-openSan text-lg font-semibold text-white w-36 h-14">
+              The North Lake Iceland
+            </p>
+            <p className="flex fles-col items-end font-openSan text-sm font-normal text-white">
```
Comment (JayceDugan) [inline src/Pages/kitchen_sink/KitchenSink.tsx]: Nit: Think the flex classes here can be removed.

File: src/Pages/kitchen_sink/KitchenSink.tsx [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+        {/* Buttons */}
+        <div className="flex flex-row justify-between mx-24 mb-16">
+          {/* Slide buttons */}
+          <div>
+            <div className="flex flex-row justify-between w-28">
+              {/* Left Button */}
+              <div className={buttonStyle} onClick={prevSlide}>
+                <FontAwesomeIcon icon={faArrowLeft} />
+              </div>
+              {/* Right Button */}
```
Comment (JayceDugan) [inline src/Pages/kitchen_sink/KitchenSink.tsx]: Great use of comments here 👏

File: src/Pages/kitchen_sink/KitchenSink.tsx [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+            </p>
+            <p className="flex fles-col items-end font-openSan text-sm font-normal text-white">
+              -16 ° C Very cold
+            </p>
+          </div>
+        </div>
+
+        {/* Buttons */}
+        <div className="flex flex-row justify-between mx-24 mb-16">
+          {/* Slide buttons */}
```
Comment (JayceDugan) [inline src/Pages/kitchen_sink/KitchenSink.tsx]: Nit & optional: Would recommend breaking these down into their own component.

File: src/Pages/kitchen_sink/KitchenSink.tsx [hunk 6]
```diff
@@ -0,0 +1,119 @@
...
+              <div className={playButtonStyle}>
+                <FontAwesomeIcon icon={faPlay} />
+              </div>
+              <div className="flex items-center font-openSan font-bold text-white text-sm">
+                <p>Watch the video</p>
+              </div>
+            </div>
+
+            {/* More info button */}
+            <div>
```
Comment (JayceDugan) [inline src/Pages/kitchen_sink/KitchenSink.tsx]: Super nitpick: Not sure if div is needed here.

File: src/Pages/kitchen_sink/KitchenSink.tsx (hunk 3 above)
Comment (JayceEse) [reply src/Pages/kitchen_sink/KitchenSink.tsx]: your right definitely didn't need them haha

File: src/Pages/kitchen_sink/KitchenSink.tsx (hunk 2 above)
Comment (JayceEse) [reply src/Pages/kitchen_sink/KitchenSink.tsx:130]: I ended up moving a lot of the div classes out to variables and putting in conditionals to change the title and description depending on which slide was showing

File: src/Pages/kitchen_sink/KitchenSink.tsx (hunk 6 above)
Comment (JayceEse) [reply src/Pages/kitchen_sink/KitchenSink.tsx]: definitely isn't needed haha

**Comment by JayceEse** [comment]
Hey @JayceDugan, 

 Just finished my homepage sliders. Just wanting your approval first before merging to master :)

File: src/Pages/home/home.tsx [hunk 7]
```diff
@@ -1,8 +1,155 @@
...
-      <div className="text-3xl w-screen h-screen flex items-center justify-center bg-yellow-300">
-        Home Section
+      <div
+        style={imgSlide}
+        className="flex flex-col justify-between duration-500 delay-75"
+      >
+        {/* Destination Info */}
+        <div
+          className={
+            currentIndex === 0 ? containerHomeClass : containerBaseClass
```
Comment (JayceDugan) [inline src/Pages/home/home.tsx]: Nit: Some duplicated code here checking if `currentIndex === 0` 

Just suggestions:

- What does currentIndex === 0 actually represent?
- Could it be a variable?
- Could you setup an object structure that made these conditional classes much easier to consume?

Comment (JayceEse) [reply src/Pages/home/home.tsx]: Yea I see what you mean haha. I think I've figured out a way to fix this going off your suggestions but not sure If I made it more complicated than it should be but Ill let you be the judge haha.

So my answer to the suggestions:
- What does currentIndex === 0 actually represent?
- Could it be a variable?
So I kinda figured out these two using this:
`const isHomeSlide = currentIndex === 0;
`
by using this variable the Homepage can see whether the first slide is showing or not

- Could you setup an object structure that made these conditional classes much easier to consume?
and for this one I created my object:
```
const classStyles = [
    {
      titleHome:
        "lg:text-[150px] md:text-9xl sm:text-8xl text-6xl font-noto text-white font-bold max-w-7xl h-auto mb-8",
      titleBase:
        "lg:text-8xl text-6xl font-noto text-white font-bold max-w-xl h-auto mb-20",
    },
    {
      paraHome:
        "font-openSan md:text-2xl text-lg font-semibold text-white max-w-4xl h-auto mb-11 mt-40",
      paraBase:
        "font-openSan text-2xl font-semibold text-white w-40 h-14 lg:mr-56 mr-20",
    },
    {
      wrapperHome: "flex-row justify-between w-auto",
      wrapperBase: "flex sm:flex-row flex-col w-auto",
    },
    {
      containerHome:
        "flex flex-col-reverse justify-between lg:ml-36 mx-12 mt-28 w-auto h-auto",
      containerBase: "lg:ml-36 mx-12 mt-40",
    },
  ];
```

and used variables to apply the appropriate styles to the homepage like this:
```
const titleStyle = isHomeSlide
    ? classStyles[0].titleHome
    : classStyles[0].titleBase;
```

**Pull request merged**